src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -1738,9 +1742,65 @@ AS
 
         -- Add the new tags with the given status
         INSERT INTO dbo.ExtendedQueryTag
-            (TagKey, TagPath, TagPrivateCreator, TagVR, TagLevel, TagStatus)
+            (TagKey, TagPath, TagPrivateCreator, TagVR, TagLevel, TagStatus, QueryStatus)
         OUTPUT INSERTED.TagKey
-        SELECT NEXT VALUE FOR TagKeySequence, TagPath, TagPrivateCreator, TagVR, TagLevel, @ready FROM @extendedQueryTags
+        SELECT NEXT VALUE FOR TagKeySequence, TagPath, TagPrivateCreator, TagVR, TagLevel, @ready, 1 FROM @extendedQueryTags
+
+    COMMIT TRANSACTION
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     UpdateExtendedQueryTagQueryStatus
+--
+-- DESCRIPTION
+--    Update QueryStatus of extended query tag
+--
+-- PARAMETERS
+--     @tagPath
+--         * The extended query tag path
+--     @queryStatus
+--         * The query  status
+--
+-- RETURN VALUE
+--     The modified extended query tag.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.UpdateExtendedQueryTagQueryStatus (
+    @tagPath VARCHAR(64),
+    @queryStatus TINYINT
+)
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    BEGIN TRANSACTION
+
+        DECLARE @tagKey INT
+        DECLARE @currentQueryStatus TINYINT
+        DECLARE @currentTagStatus TINYINT
+
+        -- Check if tag exists
+        SELECT @tagKey = TagKey, @currentQueryStatus = QueryStatus , @currentTagStatus = TagStatus
+        FROM dbo.ExtendedQueryTag WITH (HOLDLOCK)
```

Comment:
I think `UPDLOCK` is sufficient here, as we don't care if new tags are inserted while we interact with the table; the first thing we check is whether the tag is already present.